lib/compiler-singlepass/src/compiler.rs

```diff
@@ -85,82 +86,109 @@ impl Compiler for SinglepassCompiler {
             .bytes();
         let vmoffsets = VMOffsets::new(pointer_width).with_module_info(&module);
         let import_idxs = 0..module.import_counts.functions as usize;
-        let import_trampolines: PrimaryMap<SectionIndex, _> = import_idxs
-            .into_par_iter_if_rayon()
-            .map(|i| {
-                let i = FunctionIndex::new(i);
-                gen_import_call_trampoline(
-                    &vmoffsets,
-                    i,
-                    &module.signatures[module.functions[i]],
-                    calling_convention,
-                )
-            })
-            .collect::<Vec<_>>()
-            .into_iter()
-            .collect();
+        rayon::ThreadPoolBuilder::new().build().unwrap();
+        rayon::ThreadPoolBuilder::new().build().unwrap();
+        let import_trampolines: PrimaryMap<SectionIndex, _> =
+            tracing::info_span!("import_trampolines", n_imports = import_idxs.len()).in_scope(
+                || {
+                    // Note for the curious coming here after seeing a tracing trace:
+                    // I do not know either why this takes so long even with import_idxs.len() = 0
+                    // My guess is this is the place where rayon thread pools get initialized
+                    // And it happens in the .collect::<Vec<_>>() so there's no good way to add a span around it
+                    // But even adding a rayon::ThreadPoolBuilder::new().build().unwrap() above doesn't accelerate it
```

Comment:
`rayon::ThreadPoolBuilder::new().build().unwrap()` won’t register a global pool. One way to see when the threads are started would be to register a Pool’s `start_handler` to emit an event.

Reply:
… you're right I'm stupid it should be `build_global`, and this actually fixes the issue…